src/io/MuxReader.hh

```diff
@@ -96,32 +96,27 @@ public:
         }
         void reserve(size_t new_reserved) {
             if (new_reserved > mReserved) {
-                mReserved = new_reserved;
-                uint8_t *new_begin = (uint8_t*)mAlloc.allocate(mReserved);
+                uint8_t *new_begin = (uint8_t*)mAlloc.allocate(new_reserved);
                 if (mBegin != NULL) {
                     memcpy(new_begin, mBegin, mSize);
-                    mAlloc.destroy(mBegin);
+                    mAlloc.deallocate(mBegin, mReserved);
                 }
                 mBegin = new_begin;
+                mReserved = new_reserved;
             }
         }
         void resize(size_t new_size) {
             if (mReserved < new_size) {
-                mReserved *= 2;
-                if (mReserved < new_size) {
-                    mReserved = new_size;
-                }
-                uint8_t *new_begin = (uint8_t*)mAlloc.allocate(mReserved);
-                if (mBegin != NULL) {
-                    memcpy(new_begin, mBegin, mSize);
-                    mAlloc.destroy(mBegin);
+                size_t new_reserved = mReserved * 2;
+                if (new_reserved < new_size) {
+                    new_reserved = new_size;
                 }
-                mBegin = new_begin;
+                reserve(new_reserved);
             }
             dev_assert(mSize <= mReserved);
             mSize = new_size;
         }
-        ~ResizableByteBuffer() {
+        virtual ~ResizableByteBuffer() {
```

Comment:
It's inherited here: https://github.com/dropbox/lepton/blob/7533355cfc83bcb7d5a3e4947453842d41199c8b/src/lepton/stream_interfaces.hh#L8

And then I make array of unique_ptr<ResizableByteBufferListNode> in the next commit: https://github.com/interiorem/lepton/pull/3/commits/f63476ca5e4e6b51c771f56edd3962ad336e9447#diff-f5ea8c43f29affd493eb93c7ad94742cR64

So in order to automatically free the memory, we should make this destructor virtual.

Reply:
> So in order to automatically free the memory, we should make this destructor virtual.

Nope, we shouldn't, because you store pointers to `ResizableByteBufferListNode`.
Virtual inheritance is unnecessary here.